### ISSUE TITLE
LGBMRegressor with linear_tree has constant leafs

### ISSUE THREAD

**Issue by Pablo-Davila** [microsoft/LightGBM#5131]
## Description
I am trying to build a gradient boosting tree with linear models in its leafs. I am using the parameter [linear_tree](https://lightgbm.readthedocs.io/en/latest/Parameters.html#linear_tree) for this purpose.

To check wether the leafs are actually linear I have built a sample dataset and a regression model with a single estimator with the code below. The results after running the script suggest that the leafs are constant values:

![meh](https://user-images.githubusercontent.com/59798058/162086223-684408d0-8a9c-4824-b72e-56d03bb0f204.png)

Am I missing something?

## Reproducible example
```python
from lightgbm import LGBMRegressor
from sklearn.model_selection import train_test_split
import matplotlib.pyplot as plt
import numpy as np


# Sample "dataset"
x = np.arange(0,30,0.1).reshape(-1,1)
y = np.sin(x).reshape(-1)

data = np.stack([
    y[i:i+6]
    for i in range(y.shape[0] - 6)
])
x_train, x_test, y_train, y_test = train_test_split(
    data[:, :-1],
    data[:, -1],
    test_size=0.25,
    shuffle=False
)

# Model
params = {
    'objective': 'regression',
    'learning_rate': 1,
    'n_estimators': 1,
    'max_depth': 2,
    'linear_tree': True,
}
model = LGBMRegressor(**params)

model.fit(x_train, y_train)

# Plotting
plt.plot(x,y)
plt.plot(x[5:x_train.shape[0]+5].reshape(-1), model.predict(x_train))
plt.plot(x[-x_test.shape[0]:].reshape(-1), model.predict(x_test))
plt.legend(["real", "prediction (train)", "prediction (test)"])
plt.show()
```

## Environment info

LightGBM version or commit hash: v3.3.2

**Comment by Pablo-Davila** [comment]
Hi @jmoralez, thank you very much for your quick and accurate response. This definitely solves my issue.
